src/services/fileStorage/proxy-service.js

```diff
@@ -44,40 +43,95 @@ const sanitizeObj = (obj) => {
 	return obj;
 };
 
-const prepareThumbnailGeneration = (file, strategy, userId, { name: dataName }, { storageFileName, name: propName }) =>
-	ENABLE_THUMBNAIL_GENERATION
-		? Promise.all([
-				strategy.getSignedUrl({
-					userId,
-					flatFileName: storageFileName,
-					localFileName: storageFileName,
-					download: true,
-					Expires: 3600 * 24,
-				}),
-				strategy.generateSignedUrl({
-					userId,
-					flatFileName: storageFileName.replace(/(\..+)$/, '-thumbnail.png'),
-					fileType: returnFileType(dataName || propName), // data.type
-				}),
-		  ]).then(([downloadUrl, signedS3Url]) =>
-				rp
-					.post({
-						url: FILE_PREVIEW_SERVICE_URI,
-						body: {
-							downloadUrl,
-							signedS3Url,
-							callbackUrl: url.resolve(FILE_PREVIEW_CALLBACK_URI, file.thumbnailRequestToken),
-							options: {
-								width: 120,
-							},
+const getStorageProviderIdAndBucket = async (userId, fileObject, strategy) => {
+	let storageProviderId = fileObject.storageProviderId;
+	let bucket = fileObject.bucket;
+
+	// TODO: How to handle, if only one is set?
+	if (!storageProviderId) {
+		// deprecated: author check via file.permissions[0]?.refId is deprecated and will be removed in the next release
+		const creatorId =
```

Comment:
This looks like `fileObject.creator` has precedence over the user ref. Is this intended? Looks ok to me but maybe worth thinking about.

Reply:
This order of priorities is the original approach and @Loki-Afro also mentioned, that this might be required for some cases. Therefore I left it as it was.